### PR TITLE
History and keybindings in Coming from Bash

### PR DESCRIPTION
**Pull request by NotTheDr01ds**
Implemented suggestion in #1072:

* Added `!!`, `!$`, `!<n>`, and `!<-n>` history substitutions
* Also added a few Bashism keybindings and pointed to the Reedline chapter for the rest.

### CODE REVIEW

**Comment by fdncred** [comment]
You can reference this link https://www.nushell.sh/blog/2024-05-15-bashisms.html and I think there may be one missing.

**Comment by NotTheDr01ds** [comment]
Thanks - Definitely missed one.  I forgot about the blog post and was just going from what I could grok from the source ;-)

Are you saying that I should put a link near the table in the Book to the Blog post?  Or just that I could use it for my own reference?

**Comment by fdncred** [comment]
I'm fine with putting the blog information in the docs, no need to link to it.

**Comment by NotTheDr01ds** [comment]
Sounds good - Should be ready to merge if the update for `!<string>` looks good.

File: book/coming_from_bash.md [hunk 1]
```diff
@@ -69,3 +73,18 @@ $env.Path = ($env.Path | prepend 'C:\Program Files\Git\usr\bin')
 | `pwd`                                | `$env.PWD`                                                    | Display the current directory                                     |
 | `read var`                           | `let var = input`                                             | Get input from the user                                           |
 
+## History Substitutions and Default Keybindings:
+
+| Bash                                 | Nu                                                            | Task                                                              |
+| ------------------------------------ | ------------------------------------------------------------- | ----------------------------------------------------------------- |
+| `!!`                                 | `!!`                                                          | Repeat last commandline from history                              |
+| `!$`                                 | `!$`                                                          | Repeat last commandline from history                              |
```
Comment (fdncred) [inline book/coming_from_bash.md:81]: This description is wrong. `Get the last spatially separated token from the last command in your history.` from my blog post.

**Comment by NotTheDr01ds** [comment]
Thanks - Copy/paste error

File: book/coming_from_bash.md (hunk 1 above)
Comment (fdncred) [reply book/coming_from_bash.md]: Now that I look at these again, they don't repeat anything. They just get the info and put it on the command line. Bash repeats. Nushell just gets but doesn't execute.

**Comment by fdncred** [comment]
Now that I look at these again, they don't repeat anything. They just get the info and put it on the command line. Bash repeats. Nushell just gets but doesn't execute.

**Comment by NotTheDr01ds** [comment]
How about "substitute" in place of "repeat"?

**Comment by fdncred** [comment]
I think the key point that we need to make is that it does not work the same as bash. In bash you type !!<enter> and it runs the last command. I nushell you do !!<enter> and it puts the last command on the commandline but does not execute it. I'm not sure substitute or repeat convey what it does. Get is closer but it's not quite that either. Just pick one and put some blurb that describes the functionality, and I'll land it. Thanks for helping!

**Comment by NotTheDr01ds** [comment]
*"Insert"* - I think that's the word we're looking for.  Updated.

**Pull request merged**